### PR TITLE
Deployment: ERC4626 Linear Pool V3

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Deployment: 

Deployment of ERC4626 Linear Pool V3. Note that this was done from the orbcollective repo.

## Deployment tasks checklist:

- [X] Contract(s) deployed to all target networks / testnets
- [X] TX IDs are properly generated
- [X] Action IDs are generated after deployment (if applicable) 
- [X] Contracts are verified in every network 
- [X] Deployments `CHANGELOG` is updated
- [X] There are no code changes of any kind

### CODE REVIEW

File: pkg/deployments/deployment-txs/mainnet.json [hunk 1]
```diff
@@ -92,5 +92,7 @@
   "0x5Dd94Da3644DDD055fcf6B3E1aa310Bb7801EB8b": "0x39f357b78c03954f0bcee2288bf3b223f454816c141ef20399a7bf38057254c4",
   "0xdba127fBc23fb20F5929C546af220A991b5C6e01": "0xd8c9ba758cb318beb0c9525b7621280a22b6dfe02cf725a3ece0718598f260ef",
   "0x1576d472d82A72a1Ec7C57FcA770BD752D124A62": "0xed8cfdf17294c227023a1083f1d0ee385873a6458a7132e3a251ca4cd33621f0",
-  "0x222bc81C6F3C17e9e9Aba47a12f55a1Dea42f163": "0x83fb41bb8a64dc323255fc3a98d544fbd88a37e6447054c7b4dd16fc6251feac"
+  "0x222bc81C6F3C17e9e9Aba47a12f55a1Dea42f163": "0x83fb41bb8a64dc323255fc3a98d544fbd88a37e6447054c7b4dd16fc6251feac",
+  "0x67A25ca2350Ebf4a0C475cA74C257C94a373b828": "0xf252d24fcb350febba5b414dcdf93a6b08e558e384a19bf88abaf8b0dde58d14",
+  "0xE605Dbe1cA85dCdb8F43CEfA427f3B0fC06f6ba6": "0xb8151a4af682703443a331e769d6d92a04844ea2793b9dcf496811c0158d6d5b"
```
Comment (jubeira) [inline pkg/deployments/deployment-txs/mainnet.json:97]: Pool seems to be missing (so the verification is pending)

File: pkg/deployments/action-ids/arbitrum/action-ids.json [hunk 2]
```diff
@@ -550,5 +550,20 @@
...
+      "factoryOutput": "0xf984eb2B8a7eF780245a797A2fCcd82f346409ca",
+      "actionIds": {
+        "disableRecoveryMode()": "0x541412e0bfa3ff92e78aaa6272dc265e37356c9a06ea99c41e507b841cbdaf39",
+        "enableRecoveryMode()": "0x1af047704bcfd3e1466e6f18e52a32f872a6efd426a13675bc2513e447b0238d",
+        "initialize()": "0x1e783661023b1e72d6a5ee068eefcb0b8dd5b23151336636e6a0e413f2203ad8",
+        "pause()": "0x343468c5eda28b04bd0d752d84a11bd0ba70d76ed5fc481e15de7f36b4937787",
+        "setSwapFeePercentage(uint256)": "0x4e369965828f3a76cdcb37d24a69a3ddf1f0bbfa9110f9978e25d804504b279f",
+        "setTargets(uint256,uint256)": "0x342f10e709ad2a053a4e5035c4b3b703fea94c19f458d3887a6323472f0e42cd",
+        "unpause()": "0x88750d9fe81cc28abdc1c823f271c75e76cbd232145d0e83e405ce7810dd187f"
+      }
```
Comment (nventuro) [inline pkg/deployments/action-ids/arbitrum/action-ids.json:566]: We're missing the action ids for the factory (required e.g. to disable it)

**Review by jubeira — COMMENTED** [review]
Looks good.
Only mainnet pool is missing deployment / verification.

File: pkg/deployments/deployment-txs/mainnet.json (hunk 1 above)
Comment (EndymionJkb) [reply pkg/deployments/deployment-txs/mainnet.json:97]: I believe he said he ran out of ETH and didn't have enough to deploy the pool. Will look into it

**Review by rabmarut — CHANGES_REQUESTED** [review]

File: pkg/deployments/action-ids/polygon/action-ids.json [hunk 3]
```diff
@@ -635,5 +635,20 @@
         "updateTokenRateCache(address)": "0xade7055deebb6b22c267d265d1705ef0ac92207b74b405de6fe239b63380341d"
       }
     }
+  },
```
Comment (joaobrunoah) [inline pkg/deployments/action-ids/polygon/action-ids.json:638]: Replace these action IDs by:

```
"20230206-erc4626-linear-pool-v3": {
    "ERC4626LinearPoolFactory": {
      "useAdaptor": false,
      "actionIds": {
        "create(string,string,address,address,uint256,uint256,address,uint256)": "0x98933eeecdc70ae1fe94f3986354960201f8e848e98bfaf2f496681643c38541",
        "disable()": "0x5ec883ecf16774e6b304f1e06322669694ffb90c13167afeac8ed7b68ab0869e"
      }
    },
    "ERC4626LinearPool": {
      "useAdaptor": false,
      "factoryOutput": "0xf984eb2b8a7ef780245a797a2fccd82f346409ca",
      "actionIds": {
        "disableRecoveryMode()": "0x541412e0bfa3ff92e78aaa6272dc265e37356c9a06ea99c41e507b841cbdaf39",
        "enableRecoveryMode()": "0x1af047704bcfd3e1466e6f18e52a32f872a6efd426a13675bc2513e447b0238d",
        "initialize()": "0x1e783661023b1e72d6a5ee068eefcb0b8dd5b23151336636e6a0e413f2203ad8",
        "pause()": "0x343468c5eda28b04bd0d752d84a11bd0ba70d76ed5fc481e15de7f36b4937787",
        "setSwapFeePercentage(uint256)": "0x4e369965828f3a76cdcb37d24a69a3ddf1f0bbfa9110f9978e25d804504b279f",
        "setTargets(uint256,uint256)": "0x342f10e709ad2a053a4e5035c4b3b703fea94c19f458d3887a6323472f0e42cd",
        "unpause()": "0x88750d9fe81cc28abdc1c823f271c75e76cbd232145d0e83e405ce7810dd187f"
      }
    }
  }
```

File: pkg/deployments/tasks/20230206-erc4626-linear-pool-v3/output/polygon.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "ERC4626LinearPoolFactory": "0x62aaB12865d7281048c337D53a4dde9d770321E6",
```
Comment (joaobrunoah) [inline pkg/deployments/tasks/20230206-erc4626-linear-pool-v3/output/polygon.json:2]: Replace ERC4626LinearPoolFactory address by 0xa3b9515a9c557455bc53f7a535a85219b59e8b2e

File: pkg/deployments/deployment-txs/polygon.json [hunk 5]
```diff
@@ -55,5 +55,8 @@
   "0x82e4cFaef85b1B6299935340c964C942280327f4": "0x2bc079c0e725f43670898b474afedf38462feee72ef8e874a1efcec0736672fc",
   "0x7bc6C0E73EDAa66eF3F6E2f27b0EE8661834c6C9": "0xb189a45eac7ea59c0bb638b5ae6c4c93f9877f31ce826e96b792a9154e7a32a7",
   "0x93ECf386cD53718aF16a09C325a56D1bB60aDc37": "0xafd498a85b99248e957da6bdb7810d607c5e1b72bf9d69663746dfd9e0a94a90",
-  "0xeff1d57A28167E6982DCa1D63321810400f73929": "0xe088848d4502e564c261094421f6e7e159540f86f952c9ac9dc216b76c01b951"
+  "0xeff1d57A28167E6982DCa1D63321810400f73929": "0xe088848d4502e564c261094421f6e7e159540f86f952c9ac9dc216b76c01b951",
+  "0x62aaB12865d7281048c337D53a4dde9d770321E6": "0x01c76325d97b348324145f2b4608c2eb55f429fa218145499d0ee7309078b11e",
```
Comment (joaobrunoah) [inline pkg/deployments/deployment-txs/polygon.json:59]: Replace this line by 

`"0xa3B9515A9c557455BC53F7a535A85219b59e8B2E": "0x251d4ca291f258fe9c047014eced2c694db5edca2c3822ef2ecc45a177949ccc"`

**Review by joaobrunoah — APPROVED** [review]

File: pkg/deployments/tasks/20230206-erc4626-linear-pool-v3/output/mainnet.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+  "ERC4626LinearPoolFactory": "0x67A25ca2350Ebf4a0C475cA74C257C94a373b828",
+  "MockERC4626Token": "0xE605Dbe1cA85dCdb8F43CEfA427f3B0fC06f6ba6"
+}
```
[the remark this answers is not in the document]
Comment (EndymionJkb) [reply pkg/deployments/tasks/20230206-erc4626-linear-pool-v3/output/mainnet.json:5]: ? It's already there

[the remark this answers is not in the document]
Comment (rabmarut) [reply pkg/deployments/tasks/20230206-erc4626-linear-pool-v3/output/mainnet.json:5]: Confirmed, it is not showing on the web UI for some reason but it's there in the branch.

**Review by rabmarut — APPROVED** [review]

**Pull request merged**